simapp/test_util.go

```diff
@@ -21,3 +31,186 @@ func NewSimAppUNSAFE(logger log.Logger, db dbm.DB, traceStore io.Writer, loadLat
 	gapp = NewSimApp(logger, db, traceStore, loadLatest, invCheckPeriod, baseAppOptions...)
 	return gapp, gapp.keyMain, gapp.keyStaking, gapp.stakingKeeper
 }
+
+// getSimulationLog unmarshals the KVPair's Value to the corresponding type based on the
+// each's module store key and the prefix bytes of the KVPair's key.
+func getSimulationLog(storeName string, cdcA, cdcB *codec.Codec, kvA, kvB cmn.KVPair) (log string) {
```

Comment:
Hmmm, assuming the output is better when doing import/export simulation, this seems like a step in the right direction. It just seems somewhat fragile to me. What if we add new key/prefixes types? What if we add new modules? This file will need constant attention to make sure the latest changes are reflected.

Can we think of a less brittle and more modular way of doing this?

Reply:
Good point. Ideally, each module should be registering the prefix bytes of every `Key` from the store with its corresponding interface so that we could add it to the module generalization pattern.

Reply:
IMHO that shouldn't block this PR though. We could improve it on a following PR